torch_geometric/data/lightning_datamodule.py

```diff
@@ -481,11 +487,36 @@ def __init__(
 
         if loader in ['neighbor', 'link_neighbor']:
             input_type = get_edge_label_index(data, input_train_edges)[0]
-            if input_type is not None:
-                num_src_nodes = data[input_type[0]].num_nodes
-                num_dst_nodes = data[input_type[-1]].num_nodes
+
+            if isinstance(data, (Data, HeteroData)):
+                if input_type is not None:
```

Comment:
I think we can drop this logic here.

Reply:
I think we still need this, if we want to support `Data`.

Reply:
I don't think so - for `Data` and `HeteroData`, `num_src_nodes` and `num_dst_nodes` are automatically inferred, see https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/link_neighbor_loader.py#L61-L65.

Reply:
Yes, this is true in `LinkNeighborLoader`, but not in `LightningLinkData`.

Reply:
Not sure I understand, just let them be `None` instead?

Reply:
Sorry, I misunderstood. Yes, we can remove this.